src/java.base/share/classes/jdk/internal/module/ModuleInfo.java

```diff
@@ -408,24 +408,11 @@ private Builder readModuleAttribute(DataInput in, ConstantPool cpool, int major,
                     throw invalidModuleDescriptor("The requires entry for java.base"
                                                   + " has ACC_SYNTHETIC set");
                 }
-                // requires transitive java.base is illegal unless:
-                // - the major version is 53 (JDK 9), or:
-                // - the classfile is a preview classfile, or:
-                // - the module is deemed to be participating in preview
-                //   (i.e. the module is a java.* module)
                 // requires static java.base is illegal unless:
                 // - the major version is 53 (JDK 9), or:
                 if (major >= 54
-                    && ((mods.contains(Requires.Modifier.TRANSITIVE)
-                         && !isPreview
-                         && !"java.se".equals(mn))
-                        || mods.contains(Requires.Modifier.STATIC))) {
-                    String flagName;
-                    if (mods.contains(Requires.Modifier.STATIC)) {
-                        flagName = "ACC_STATIC_PHASE";
-                    } else {
-                        flagName = "ACC_TRANSITIVE";
-                    }
+                    && mods.contains(Requires.Modifier.STATIC)) {
+                    String flagName = "ACC_STATIC_PHASE";
```

Comment:
A minor comment on the style here, do you mind combin L413+414 as it will make it easier to read. The change is okay of course.

Reply:
I did this:
https://github.com/openjdk/jdk/pull/23801/commits/029e0dd59c7160f4fe043dd7191860b067ca787c
Please let me know if something more/different is needed.